espsecure/esp_hsm_sign/__init__.py

```diff
@@ -36,6 +37,12 @@ def read_hsm_config(configfile):
         if not config.has_option(section, option):
             raise configparser.NoOptionError(option, section)
 
+    # Check if config.["credentials"] is set to "prompt" and prompt for HSM PIN
+    credentials_config_val = config.get(section, "credentials")
+    if credentials_config_val == "prompt":
+        hsm_pin = getpass("\nPlease enter the PIN of your HSM:\n")
+        config.set(section, "credentials", hsm_pin)
+
```

Comment:
Alternatively, rather than using the word "prompt" (might be misleading), we can just remove the `credentials` option from the config file and add a check that if the option is absent, get the credentials using a prompt.

```suggestion
    # If the config file does not contain the "credentials" option, prompt for the HSM PIN
    if not config.has_option(section, "credentials"):
        hsm_pin = getpass("Please enter the PIN of your HSM:\n")
        config.set(section, "credentials", hsm_pin)

```

Reply:
Could you also add a note in the [documentation](https://github.com/espressif/esptool/blob/master/docs/en/espsecure/index.rst#hsm-config-file) regarding these changes?
Something like,
```
To avoid storing the PIN in plaintext, you could simply remove the `credentials` option from the config file, and during signing using the HSM, you could enter it using the prompt.
```
would suffice.

Reply:
For documentation update (if we stay with prompt) :  This is what I have in mind:

1. Change the sample to say `credentials = prompt` and then document 
```
Below is a sample HSM config file (hsm_config.ini) for using [SoftHSMv2](https://github.com/opendnssec/SoftHSMv2) as an external HSM:

>> insert config file with credentials = prompt here as in the current example <<

Using this config file, the script will prompt for the HSM pin to be typed in for operations that need HSM access.
Alternatively the keyword `prompt` can be replaced with the `HSM pin` (e.g. 1234) to avoid being prompted.
Note that this is not a good practice for security and using `prompt` is recommended.
```

This way the example (which most people will copy-and paste) already sets them up to the `safer behavior`

Reply:
@rretanubun There might be a case where the user pin of a token could be the word "prompt" itself (tried using SoftHSM2, it seems possible). In such a situation the user will not be able to get the expected behavior.
Even if say we do not come across such a situation, I think making the `credentials` field optional would be a better approach rather than deciding on a word (say "prompt") and hardcoding it into the tool.

Reply:
Fair point. Code and documentation updated as requested.